src/map/ai/states/magic_state.cpp

```diff
@@ -182,7 +182,7 @@ bool CMagicState::CanCastSpell(CBattleEntity* PTarget)
     {
         return false;
     }
-    if (distance(m_PEntity->loc.p, PTarget->loc.p) > 40)
```

Comment:
At any rate, Teo is right in this regard - the mob range check is right after the spell range check.